semantic-source/src/Source/Source.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE DeriveGeneric, GeneralizedNewtypeDeriving #-}
+{-|
+'Source' models source code, represented as a thin wrapper around a 'B.ByteString' with conveniences for splitting by line, slicing, etc.
+
+This module is intended to be imported qualified to avoid name clashes with 'Prelude':
```

Comment:
We dropped prefixes from this module (🎩 @patrickt!) in keeping with the style guide.